src/main/java/seedu/address/logic/parser/OrganiseCommandParser.java

```diff
@@ -0,0 +1,156 @@
+     * @throws ParseException if the user input does not conform the expected format.
+     */
+    private OrganiseCommand parseRecommendation(String args) throws ParseException {
+        try {
+            ContactIndex contactIndex = new ContactIndex(Integer.parseInt(args.trim()));
+            return new OrganiseCommand(contactIndex);
+        } catch (NumberFormatException nfe) {
+            throw new ParseException(MESSAGE_INVALID_COMMAND_FORMAT);
+        }
+    }
+
+    /**
+     * Parses user input to get meeting participants.
+     * @param argumentMultimap Mappings of prefixes to their respective arguments.
+     * @return Participants of the meet up.
+     * @throws ParseException if the user input does not conform the expected format.
+     */
+    private Participants parseParticipants(ArgumentMultimap argumentMultimap) throws ParseException {
+        List<String> indexArray = Arrays.stream(argumentMultimap.getPreamble().split(" "))
+                .filter(x -> !x.isEmpty())
+                .collect(Collectors.toList());
+
+        Set<ContactIndex> indices = ParserUtil.parseIndices(indexArray);
+        if (indices.isEmpty()) {
+            throw new ParseException(MESSAGE_NO_PARTICIPANTS_GIVEN);
+        }
+
+        Participants participants = new Participants(indices);
+        return participants;
+    }
+
+    /**
+     * Parses user input to get meeting day.
+     * @param argumentMultimap Mappings of prefixes to their respective arguments.
+     * @return Day of the meet up.
+     * @throws ParseException if the user input does not conform the expected format.
+     */
+    private Day parseDay(ArgumentMultimap argumentMultimap) throws ParseException {
+        if (!argumentMultimap.getValue(Prefix.DAY).isPresent()) {
+            throw new ParseException(MESSAGE_NO_DATE_GIVEN);
+        }
+        Day day;
+        try {
+            day = Day.valueOf(argumentMultimap.getValue(Prefix.DAY).get());
+        } catch (IllegalArgumentException e) {
+            throw new ParseException(MESSAGE_WRONG_DATE_FORMAT);
+        }
+        return day;
+    }
+
+    /**
+     * Parses user input to get time period of the meeting.
+     * @param argumentMultimap Mappings of prefixes to their respective arguments.
+     * @return Time period of the meet up.
+     * @throws ParseException if the user input does not conform the expected format.
+     */
+    private TimePeriod parseTimePeriod(ArgumentMultimap argumentMultimap) throws ParseException {
+        List<String> time = List.of(argumentMultimap.getValue(Prefix.TIME).get().split(" "));
+        LocalTime startHour;
+        LocalTime endHour;
+        try {
+            startHour = new LocalTime(Integer.parseInt(time.get(0)), 0);
+            endHour = new LocalTime(Integer.parseInt(time.get(1)), 0);
+        } catch (NumberFormatException | IllegalFieldValueException nfe) {
+            throw new ParseException(MESSAGE_WRONG_TIME_FORMAT);
```

Comment:
FYI, can use ParserUtil for this. Helps you validate input also.

Reply:
We keep this for v1.4 yah?

Reply:
```java
public static LocalTime parseStartHour(int hour) throws ParseException```

and 
```java
public static LocalTime parseEndHour(int hour) throws ParseException
```

Reply:
Alright, changed!